### ISSUE TITLE
General approach for an ignoreErrors solution

### ISSUE THREAD

**Issue by TiMESPLiNTER** [phpmd/phpmd#886]
## Description
Currently the only way to ignore warnings/errors detected by PHPMD is by adding an annotation in the code file. (Like proposed in #515 and #180.)
I would really appreciate a solution like for example PHPStan has where I can put under a section `ignoreErrors` a list of errors I want to ignore and not cause PHPMD to exit with a non-zero exit code nor listing them in a report. Why? Because I don't want to put PHPMD specific annotations into my code files but preferably do all the project configuration regarding PHPMD in the same config file.

For example something like this:

**/phpmd.xml**
```
<ruleset name="phpmd"
         xmlns="http://pmd.sf.net/ruleset/1.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"
         xsi:noNamespaceSchemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd">
    <description>PHPMD Rules</description>

    <rule ref="rulesets/unusedcode.xml"/>
    <ignoredErrors>
        <ignoredError pattern="@.*src/PushProcessor\.php:137   Avoid unused parameters such as '\$message'\.$@" />
    </ignoreErrors>
</ruleset>

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [ ] Tell if you have the option to provide the code for this proposal.
* [x] This issue is about 1 feature proposal and nothing more.
* [x] The issue has a descriptive title. For example: "Add JSON render option".

**Comment by kylekatarnls** [comment]
@JeroenHeemskerk if what you need is to completely disable a rule for the whole project it's already possible by config:
https://phpmd.org/documentation/creating-a-ruleset.html#excluding-rules-from-a-rule-set

```xml
<?xml version="1.0"?>
<ruleset name="My first PHPMD rule set"
         xmlns="http://pmd.sf.net/ruleset/1.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0
                     http://pmd.sf.net/ruleset_xml_schema.xsd"
         xsi:noNamespaceSchemaLocation="
                     http://pmd.sf.net/ruleset_xml_schema.xsd">
    <rule ref="rulesets/naming.xml">
        <exclude name="ShortVariable" />
        <exclude name="LongVariable" />
    </rule>
</ruleset>
```

**Comment by AJenbo** [comment]
This is addressed by the new baseline feature I would say (2.10  Apr 26, 2021 )
